tools/config.schema.yaml

```diff
@@ -0,0 +1,855 @@
+            $ref: "#/definitions/PROVIDER"
+
+  PHONE_ACTIVITY_RECOGNITION:
+    type: object
+    required: [TABLE, EPISODE_THRESHOLD_BETWEEN_ROWS, PROVIDERS]
+    properties:
+      TABLE:
+        type: object
+        required: [ANDROID, IOS]
+        properties:
+          ANDROID: 
+            type: string
+          IOS: 
+            type: string
+      EPISODE_THRESHOLD_BETWEENROWS:
+        type: integer
+        maximum: 5
+        minimum: 0
+      PROVIDERS:
+        type: ["null", object]
+        properties:
+          RAPIDS:
+            allOf:
+              - $ref: "#/definitions/PROVIDER"
+              - properties:
+                  FEATURES:
+                    uniqueItems: True
+                    items:
+                      type: string
+                      enum: ["count", "mostcommonactivity", "countuniqueactivities", "durationstationary", "durationmobile", "durationvehicle"]
+                  ACTIVITY_CLASSES:
+                    type: object
+                    required: [STATIONARY, MOBILE, VEHICLE]
+                    properties:
+                      STATIONARY: 
+                        type: array
+                        uniqueItems: True
+                        items:
+                          type: string
+                          enum: ["still", "tilting"]
+                      MOBILE: 
+                        type: array
+                        uniqueItems: True
+                        items:  
+                          type: string
+                          enum: ["on_foot", "walking", "running", "on_bicycle"]
+                      VEHICLE:
+                        type: array
+                        uniqueItems: True
+                        items:
+                          type: string
+                          enum: ["in_vehicle"]
+        additionalProperties: 
+            $ref: "#/definitions/PROVIDER"
+        
+
+
+
+  PHONE_APPLICATIONS_CRASHES:
+    type: object
+    required: [TABLE, APPLICATION_CATEGORIES, PROVIDERS]
+    properties:
+      TABLE: 
+        type: string
+      APPLICATION_CATEGORIES:
+        allOf:
+          - $ref: "#/definitions/APPLICATION_CATEGORIES_FEATURE"
+      PROVIDERS:
+        additionalProperties: 
+            $ref: "#/definitions/PROVIDER"
+
+  PHONE_APPLICATIONS_FOREGROUND:
+    type: object
+    required: [TABLE, APPLICATION_CATEGORIES, PROVIDERS]
+    properties:
+      TABLE:
+        type: string
+      APPLICATION_CATEGORIES:
+        allOf:
+          - $ref: "#/definitions/APPLICATION_CATEGORIES_FEATURE"
+      PROVIDERS:
+        type: ["null", object]
+        properties:
+          RAPIDS: 
+            allOf:
+              - $ref: "#/definitions/PROVIDER"
+              - properties:
+                  SINGLE_CATEGORIES: 
+                    type: array
+                    uniqueItems: True
+                    items:
+                      type: string
+                      enum: ["all", "email"]
```

Comment:
This one shouldn't be an enum, the name of the categories are arbitrary

Reply:
fixed